### ISSUE TITLE
album video custom filter no sound

### ISSUE THREAD

**Issue by henvyluk** [BradLarson/GPUImage2#205]
I read the album by reading the video file, after a custom filter, the video file is exported without sound, very confused, for the time being did not find an effective solution to the problem, I hope you can help or tips, the following is my relevant code :

//Render display
renderView = RenderView(frame: CGRect(x: 0, y: UIScreen.Screen().height*0.08+0.5, width: UIScreen.Screen().width, height: UIScreen.Screen().width))
            view.addSubview(renderView)
movie = try MovieInput(url: url, playAtActualSpeed: true, loop: true)
            movieFilter = Normalfilter()
            movie --> movieFilter --> renderView
            movie.start()

//Sandbox directory address
tmpFileURL = NSURL.fileURL(withPath: "\(NSTemporaryDirectory())tmp\(arc4random()).mp4")
            print(tmpFileURL)
 //Export    
            output = try! MovieOutput(URL: tmpFileURL, size: Size(width: Float(UIScreen.main.bounds.height), height: Float(UIScreen.main.bounds.width)))
            movie --> movieFilter --> output
            output.startRecording()

Finally, the export of the video file is no sound, is where I have a problem?

**Comment by henvyluk** [comment]
@gillygize @wallerdev @mz2 @datskos @rounak @SoundandStuff

**Comment by zubco** [comment]
@henvyluk did you tried, to set camera.audioEncodingTarget ?
`camera.audioEncodingTarget = output`

**Comment by zubco** [comment]
My previous commend was a misunderstanding, sorry :|

**Comment by zubco** [comment]
@henvyluk In MovieInput is a commend from @BradLarson:
`// TODO: Audio here`
So, I guess there is no sound setup for MovieInput component yet.

**Comment by henvyluk** [comment]
Yes, I have noticed. I also have access to relevant information, try to do this, but not yet achieved. I have seen someone to put forward the demand, it was also a Mac version [https: //github.com/BradLarson/GPUImage2/pull/140!],  but the iOS version I was trying to be unsuccessful

**Comment by henvyluk** [comment]
Yes, I have noticed. I also have access to relevant information, try to do this, but not yet achieved. I have seen someone to put forward the demand, it was also a Mac version [https: //github.com/BradLarson/GPUImage2/pull/140!](https: //github.com/BradLarson/GPUImage2/pull/140), but the iOS version I was trying to be unsuccessful.

**Comment by henvyluk** [comment]
Yes, I have noticed. I also have access to relevant information, try to do this, but not yet achieved. I have seen someone to put forward the demand, it was also a Mac version
 [https: //github.com/BradLarson/GPUImage2/pull/140] (https: //github.com/BradLarson/GPUImage2/pull/140) , but the iOS version I was trying to be unsuccessful.

**Comment by henvyluk** [comment]
Yes, I have noticed. I also have access to relevant information, try to do this, but not yet achieved. I have seen someone to put forward the demand, it was also a Mac version
(https: //github.com/BradLarson/GPUImage2/pull/140) , but the iOS version I was trying to be unsuccessful.
